src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8351,7 +8353,7 @@ public override BoundNode VisitYieldReturnStatement(BoundYieldReturnStatement no
             {
                 return null;
             }
-            var method = _delegateInvokeMethod ?? (MethodSymbol)CurrentSymbol;
+            var method = (MethodSymbol)CurrentSymbol;
```

Comment:
I didn't quite understand why the handling of `return` and `yield return` are diverging. What's the rationale?

Reply:
`Yield return` is illegal in a lambda. This code was always wrong.

Reply:
Ah, makes sense. Thanks